scalalib/src/GenIdeaImpl.scala

```diff
@@ -443,27 +449,40 @@ case class GenIdeaImpl(
 
     val libraries: Seq[(SubPath, Elem)] =
       resolvedLibraries(allResolved).flatMap { resolved =>
-        import resolved.path
         val names = libraryNames(resolved)
         val sources = resolved match {
           case CoursierResolved(_, _, s) => s
           case WithSourcesResolved(_, s) => s
           case OtherResolved(_) => None
         }
         for (name <- names)
-          yield Tuple2(
-            os.sub / "libraries" / s"${ideaifyLibraryName(name)}.xml",
-            libraryXmlTemplate(
-              name = name,
-              path = path,
-              sources = sources,
-              scalaCompilerClassPath =
-                librariesProperties.getOrElse(path, Loose.Agg.empty)
+          yield {
+            val compilerCp: Agg[Path] = librariesProperties.getOrElse(resolved.path, Agg.empty)
+            val languageLevel = name match {
+              case _ if compilerCp.iterator.isEmpty => None
+              case _ if name.startsWith("scala3-library_3-3.0.") => Some("Scala_3_0")
+              case _ if name.startsWith("scala-library-2.13.") => Some("Scala_2_13")
+              case _ if name.startsWith("scala-library-2.12.") => Some("Scala_2_12")
+              case _ if name.startsWith("scala-library-2.11.") => Some("Scala_2_11")
+              case _ if name.startsWith("scala-library-2.10.") => Some("Scala_2_10")
+              case _ if name.startsWith("scala-library-2.9.") => Some("Scala_2_9")
+              case _ if name.startsWith("dotty-version-0.27") => Some("Scala_0_27")
+              case _ => None
```

Comment:
Decided to hardcode the mapping, as Scala 3 support in IJ is still in development and it's a bit unclear if we settle with the current mapping or not.